### ISSUE TITLE
Experiment: CDS from different evolutionary timescales

### ISSUE THREAD

**Issue by gonzalobenegas** [Open-Athena/bolinas-dna#58]
## Description

Similar to #55 but for CDS.

Perhaps train for 17k steps rather than a full epoch.

## Links

[Training code](https://github.com/marin-community/marin/blob/ea285e49adb1f3ed06397f33e009ad39b13e72ab/experiments/dna/exp58_evolutionary_timescales_cds.py)
[Analysis code](https://github.com/Open-Athena/bolinas-dna/tree/main/snakemake/analysis/evals_v1)
Wandb: [mammals](https://wandb.ai/gonzalobenegas/marin/runs/exp58-mammals-r01-23d451?nw=nwusergonzalobenegas), [vertebrates](https://wandb.ai/gonzalobenegas/marin/runs/exp58-vertebrates-r01-771c3a?nw=nwusergonzalobenegas), [animals](https://wandb.ai/gonzalobenegas/marin/runs/exp58-animals-r01-1e3682?nw=nwusergonzalobenegas)

## Results

Eval: TraitGym Mendelian traits v2 (validation set, only odd chromosomes).
- For missense variants, shorter timescales work better initially but the longer timescales surpass them in the end. Animals seems to be the optimal one.
- For synonymous and splicing variants, there seems to be less differences between models.
<img width="2188" height="576" alt="Image" src="https://github.com/user-attachments/assets/afa5c281-94ad-4f12-8319-34118603d003" />

**Comment by gonzalobenegas** [comment]
🤖 Raw AUPRC numbers behind the plot above, assembled from the [evals_v1 pipeline outputs](https://github.com/Open-Athena/bolinas-dna/tree/main/snakemake/analysis/evals_v1) in S3 (`s3://oa-bolinas/snakemake/analysis/evals_v1/results/metrics/traitgym_mendelian_v2/exp58-*/`).

**File:** [exp58_auprc_traitgym_mendelian_v2.parquet](https://gist.githubusercontent.com/gonzalobenegas/34dded27de006485839006f8349980be/raw/3fa4c0b364f781459775b271180f4cfa9440fb7f/exp58_auprc_traitgym_mendelian_v2.parquet)

**Shape:** 1836 rows = 3 models × 17 checkpoint steps × 9 subsets × 4 score_types.

**Columns:**
- `model` — `mammals` / `vertebrates` / `animals`
- `model_run` — full training-run name (e.g. `exp58-mammals-r01-23d451`)
- `step` — checkpoint step
- `metric` — `AUPRC` (the only metric computed for this dataset)
- `score_type` — `minus_llr`, `abs_llr`, `embed_last_l2`, `embed_middle_l2`
- `subset` — `global`, `tss_proximal`, `5_prime_UTR_variant`, `3_prime_UTR_variant`, `non_coding_transcript_exon_variant`, `missense_variant`, `synonymous_variant`, `splicing`, `distal`
- `value` — AUPRC
- `n_pos`, `n_neg` — positives / negatives in that subset

Eval dataset: [`bolinas-dna/evals-traitgym_mendelian_v2`](https://huggingface.co/datasets/bolinas-dna/evals-traitgym_mendelian_v2) (validation set, odd chromosomes). The plot in the issue uses `score_type == 'minus_llr'` and `subset in {'missense_variant', 'synonymous_variant', 'splicing'}`.

```python
import pandas as pd
df = pd.read_parquet('https://gist.githubusercontent.com/gonzalobenegas/34dded27de006485839006f8349980be/raw/3fa4c0b364f781459775b271180f4cfa9440fb7f/exp58_auprc_traitgym_mendelian_v2.parquet')
```

**Comment by gonzalobenegas** [comment]
🤖 Updated plot using the post-#194 evals_v2 dataset (`mendelian_traits`, hf_revision `4aed58e`).

![exp58_evolutionary_timescales_auprc](https://gist.githubusercontent.com/gonzalobenegas/3649e68fb63ca1f3443e4486078eb4d8/raw/c785daa5a1a938365ace6861791fa3e978ac53cf/exp58_evolutionary_timescales_auprc.svg)

Recipe: [`plots/exp58_evolutionary_timescales.py`](https://github.com/Open-Athena/marin-dna/blob/0058ec4779401307a9b3a3c9218127511711bce4/plots/exp58_evolutionary_timescales.py)
